ports/mnn/portfile.cmake

```
@@ -0,0 +1,105 @@
+vcpkg_fail_port_install(ON_ARCH "arm" ON_TARGET "uwp" "ios" "android")
+if(VCPKG_TARGET_IS_WINDOWS)
+  vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
+endif()
+
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO alibaba/MNN
+    REF 1.1.0
+    SHA512 3e31eec9a876be571cb2d29e0a2bcdb8209a43a43a5eeae19b295fadfb1252dd5bd4ed5b7c584706171e1b531710248193bc04520a796963e2b21546acbedae0
+    HEAD_REF master
+    PATCHES
+        use-package-and-install.patch
+)
+
+vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
+  FEATURES
+    test        MNN_BUILD_TEST
+    test        MNN_BUILD_BENCHMARK
+    cuda        MNN_CUDA
+    cuda        MNN_GPU_TRACE
+    vulkan      MNN_VULKAN
+    vulkan      MNN_GPU_TRACE
+    vulkan      MNN_USE_SYSTEM_LIB
+    opencl      MNN_OPENCL
+    opencl      MNN_USE_SYSTEM_LIB
+    metal       MNN_METAL
+    metal       MNN_GPU_TRACE
+    tools       MNN_BUILD_TOOLS
+    tools       MNN_BUILD_QUANTOOLS
+    tools       MNN_BUILD_TRAIN
+    tools       MNN_EVALUATION
+    tools       MNN_BUILD_CONVERTER
+)
+
+# 'cuda' feature in Windows failes with Ninja because of parallel PDB access. Make it optional
+set(NINJA_OPTION PREFER_NINJA) 
+if("cuda" IN_LIST FEATURES)
+    unset(NINJA_OPTION)
+endif()
+
+if(VCPKG_TARGET_IS_WINDOWS)
+    string(COMPARE EQUAL "${VCPKG_CRT_LINKAGE}" "static" USE_RUNTIME_MT)
+    list(APPEND PLATFORM_OPTIONS -DMNN_WIN_RUNTIME_MT=${USE_RUNTIME_MT})
+endif()
+
+string(COMPARE EQUAL "${VCPKG_LIBRARY_LINKAGE}" "dynamic" BUILD_SHARED)
+
+vcpkg_configure_cmake(
+    SOURCE_PATH ${SOURCE_PATH}
+    ${NINJA_OPTION}
+    OPTIONS
+        ${FEATURE_OPTIONS} ${PLATFORM_OPTIONS}
+        -DMNN_BUILD_SHARED_LIBS=${BUILD_SHARED}
+        # 1.1.0.0-${commit}
+        -DMNN_VERSION_MAJOR=1 -DMNN_VERSION_MINOR=1 -DMNN_VERSION_PATCH=0 -DMNN_VERSION_BUILD=0 -DMNN_VERSION_SUFFIX=-d6795ad
+    OPTIONS_DEBUG
+        -DMNN_DEBUG_MEMORY=ON -DMNN_DEBUG_TENSOR_SIZE=ON
+)
+vcpkg_install_cmake()
+vcpkg_copy_pdbs()
+
+vcpkg_download_distfile(COPYRIGHT_PATH
+    URLS "https://apache.org/licenses/LICENSE-2.0.txt" 
+    FILENAME copyright

```

```suggestion
    FILENAME 98f6b79b778f7b0a1541.txt
```
This filename needs to be unique, since it's shared/cached inside downloads (`downloads/$filename`)

---

A ha... nice behavior. I like it :D